src/main/app/controller/courts/CourtTypesController.ts

```diff
@@ -0,0 +1,153 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {CourtType, CourtTypeItem, CourtTypePageData} from '../../../types/CourtType';
+import {CSRF} from '../../../modules/csrf';
+
+
+@autobind
+export class CourtTypesController {
+
+
+  emptyCourtCodeErrorMsg = 'Court code is required and must be numeric.';
+  getCourtTypesErrorMsg = 'A problem occurred when retrieving the list of court types.';
+  updateErrorMsg = 'A problem occurred when saving the court court types.';
+  emptyCourtTypesErrorMsg = 'One or more court types are required.';
+
+
+  public async get(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    courtCourtTypes: CourtType[] = null): Promise<void> {
+    if (!courtCourtTypes) {
+
+      const slug: string = req.params.slug as string;
+      await req.scope.cradle.api.getCourtCourtTypes(slug)
+        .then((value: CourtType[]) => courtCourtTypes = value)
+        .catch(() => error += this.getCourtTypesErrorMsg);
+
+    }
+
+    let allCourtTypes: CourtType[] = [];
+
+    await req.scope.cradle.api.getCourtTypes()
+      .then((value: CourtType[]) => allCourtTypes = value)
+      .catch(() => error += this.getCourtTypesErrorMsg);
+
+
+    const pageData: CourtTypePageData = {
+      errorMsg: error,
+      updated: updated,
+      items: this.mapCourtTypeToCourtTypeItem(allCourtTypes, courtCourtTypes)
+
+    };
+
+    res.render('courts/tabs/typesContent', pageData);
+  }
+
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+
+    let courtCourtTypes: CourtType[] = null;
+
+    if(!CSRF.verify(req.body._csrf)) {
+      return this.get(req, res, false, this.updateErrorMsg, courtCourtTypes);
+    }
+
+    if (req.body.types) {
+
+      courtCourtTypes = this.mapBodyToCourtType(req.body);
+
+      if(courtCourtTypes.find( c => (c.name ==="Magistrates' Court" && this.CheckCodeIsNullOrNan(c.code) )
+      || (c.name==='County Court' && this.CheckCodeIsNullOrNan(c.code))
```

Comment:
Do you think it would be possible to not match on the name string? I'm slightly concerned that if someone changes the name of a court type in future, the portal behaviour will break.

Reply:
using ID  to match now